### PR TITLE
[ADD] Add mail_attach_existing_attachment module

### PR DESCRIPTION
**Pull request by adrienpeiffer**
This module was written to add the possibility to add attachments located on the object by sending it by email with the mail compose message wizard

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I think a more descriptive name can be _mail_attach_existing_attachment_.

Another functional question. Why limiting to the attachments of the object? It would be also good to attach any of the existing attachments in the system, don't you think?

It would be good also to have a Usage section explaining that the existing object attachments will be proposed to be included in the mail compose windows through checkboxes.

File: mail_attach_attachment/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#
+##############################################################################
+{
+    'name': "Mail Attach Attachment",
+    'summary': """
+        Adding attachment on the object by sending this one""",
+    'author': "ACSONE SA/NV,Odoo Community Association (OCA)",
+    'website': "http://acsone.eu",
+    'category': 'Uncategorized',
+    'version': '8.0.0.0.1',
```
Comment (pedrobaeza) [inline mail_attach_attachment/__openerp__.py]: 8.0.1.0.0

**Comment by sbidoul** [comment]
@pedrobaeza I would leave the name as is, knowing that adding the possibility to add other existing attachments can be added in the same module in another PR.

For now the module is useful as is and covers a useful use case: when the user need to send annexes to an invoices or a sale/purchase order, he can attach the annexes to the object and send them easily later by clicking check boxes in the send email dialog. 

@adrienpeiffer I suggest you add a screenshot in the README.rst. There is an example on how to do this in the mis_builder PR.

**Comment by pedrobaeza** [comment]
To add "existing" word is only to clear more the purpose of the module and serves also in case of any future enhancement.

About the enhancement itself, it's enough to put it on the "Known issues / Roadmap" section.

**Comment by sbidoul** [comment]
@pedrobaeza ok to add "existing" in the module name.

**Comment by adrienpeiffer** [comment]
Hi @pedrobaeza , @sbidoul ,
I modified the readme file, added screenshots, changed the version numbering and renamed the module.
Thanks for your reviews !

**Comment by rafaelbn** [comment]
@adrienpeiffer I think this module must install module "document" if not, how you can attach a new document? I have tested in runbot, and If I don't install document modules I cannot reproduce the functionality. Let me know you opinion.

**Comment by adrienpeiffer** [comment]
@rafaelbn Yes, technically the module can run without it. However, to add attachments, we need the document module. I'll add it.

**Comment by rafaelbn** [comment]
:+1:  tested in runbot.

File: mail_attach_existing_attachment/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+This module was written to add the possibility to add attachments located on 
+the object by sending it by email with the mail compose message wizard
+
+Installation
+============
+
+To install this module, you need to:
+
+ * Click on install
```
Comment (pedrobaeza) [inline mail_attach_existing_attachment/README.rst]: Don't put initial space

File: mail_attach_existing_attachment/README.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Configuration and Usage
```
Comment (pedrobaeza) [inline mail_attach_existing_attachment/README.rst]: This is only "Usage" IMO

File: mail_attach_existing_attachment/README.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+* Then, by sending the object via email, you can select the attachment added earlier
+
+.. figure:: static/description/ex_mail_compose_message.png
+   :alt: Sends the Purchase Order by email
+
+Known issues / Roadmap
+======================
+
+The module only allows the addition of attachments linked to the object.
```
Comment (pedrobaeza) [inline mail_attach_existing_attachment/README.rst]: Put `*` at the beginning

File: mail_attach_existing_attachment/README.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+* Adding some attachments on an object 
+
+.. figure:: static/description/attachment.png
+   :alt: Attachment on purchase order
+
+* Then, by sending the object via email, you can select the attachment added earlier
+
+.. figure:: static/description/ex_mail_compose_message.png
+   :alt: Sends the Purchase Order by email
```
Comment (pedrobaeza) [inline mail_attach_existing_attachment/README.rst:30]: You can add the new Try me on runbot button:

```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/205/8.0
```

File: mail_attach_existing_attachment/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': "Mail Attach Existing Attachment",
+    'summary': """
+        Adding attachment on the object by sending this one""",
+    'author': "ACSONE SA/NV,Odoo Community Association (OCA)",
+    'website': "http://acsone.eu",
+    'category': 'Uncategorized',
```
Comment (pedrobaeza) [inline mail_attach_existing_attachment/__openerp__.py]: I would suggest the same of mail module: `Social Network`

**Comment by pedrobaeza** [comment]
Some remarks on the README, but :+1: functional and code review

**Comment by adrienpeiffer** [comment]
@pedrobaeza it's done ! Thanks !

**Comment by pedrobaeza** [comment]
:+1:

**Comment by hbrunn** [comment]
:+1:

**Pull request merged**